### PR TITLE
rev coredns and etcd version for upgrade test

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind failing-test

**What this PR does / why we need it**:

This PR updates the version of coredns used during the CAPI workload cluster upgrade test to match the current version in CAPI main branch.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/config/azure-dev.yaml [hunk 1]
```diff
@@ -194,7 +194,7 @@ variables:
   KUBERNETES_VERSION_API_UPGRADE_FROM: "v1.26.12"
   FLATCAR_VERSION: "${FLATCAR_VERSION:-latest}"
   ETCD_VERSION_UPGRADE_TO: "3.5.4-0"
-  COREDNS_VERSION_UPGRADE_TO: "v1.9.3"
+  COREDNS_VERSION_UPGRADE_TO: "v1.11.1"
```
Comment (nojnhuh) [inline test/e2e/config/azure-dev.yaml:197]: :+1: matches CAPI on the release-1.6 branch: https://github.com/kubernetes-sigs/cluster-api/blob/90d78fa0c57abacb88a9d158c61a47fe24cda229/test/e2e/config/docker.yaml#L322

Should we maybe bump etcd to `3.5.10-0` too?

Comment (jackfrancis) [reply test/e2e/config/azure-dev.yaml:197]: Good idea.

Ace detective work @nojnhuh!

**Review by nojnhuh — APPROVED** [review]
/lgtm
/approve

/hold to let the optional upgrade test finish.

**Comment by nojnhuh** [comment]
upgrade job passed!

/hold cancel

**Comment by nojnhuh** [comment]
/cherry-pick release-1.14

I think this is holding up #4856.

**Pull request merged**